### PR TITLE
[nn] Add support for +=, * and *= operations for nn.Sequential objects

### PR DESCRIPTION
**Pull request by anshradh**
Fixes 71329

### CODE REVIEW

File: torch/nn/modules/container.py [hunk 1]
```diff
@@ -135,6 +135,59 @@ def __add__(self, other) -> 'Sequential':
...
+            combined = Sequential()
+            offset = 0
+            for _ in range(other):
+                for module in self:
+                    combined.add_module(str(offset), module)
+                    offset += 1
+            return combined
+
+    def __rmul__(self, other: int) -> 'Sequential':
+        if not isinstance(other, int):
```
Comment (albanD) [inline torch/nn/modules/container.py:162]: This can re-use `__mul__` above to avoid duplication

File: torch/nn/modules/container.py [hunk 2]
```diff
@@ -135,6 +135,59 @@ def __add__(self, other) -> 'Sequential':
...
+            raise TypeError(f"unsupported operand type(s) for *: {type(self)} and {type(other)}")
+        elif (other <= 0):
+            raise ValueError(f"Non-positive multiplication factor {other} for {type(self)}")
+        else:
+            offset = len(self)
+            copy = Sequential()
+            for i, module in enumerate(self):
+                copy.add_module(str(i), module)
+            for _ in range(other-1):
+                for module in copy:
```
Comment (albanD) [inline torch/nn/modules/container.py:186]: You don't actually need the copy here right? You can simply loop from 0 to offset every time

**Review by albanD — COMMENTED** [review]
Sounds ok.
Small comments for the implementation to simplify things!

File: torch/nn/modules/container.py (hunk 2 above)
Comment (anshradh) [reply torch/nn/modules/container.py]: Agreed.

File: torch/nn/modules/container.py (hunk 1 above)
Comment (anshradh) [reply torch/nn/modules/container.py]: Done!

File: torch/nn/modules/container.py [hunk 3]
```diff
@@ -135,6 +135,46 @@ def __add__(self, other) -> 'Sequential':
...
+            combined = Sequential()
+            offset = 0
+            for _ in range(other):
+                for module in self:
+                    combined.add_module(str(offset), module)
+                    offset += 1
+            return combined
+
+    def __rmul__(self, other: int) -> 'Sequential':
+        return self.mul(other)
```
Comment (albanD) [inline torch/nn/modules/container.py:162]: Shouldn't this be `self.__mul__(other)` or `self * other` ?

File: torch/nn/modules/container.py [hunk 4]
```diff
@@ -135,6 +135,46 @@ def __add__(self, other) -> 'Sequential':
...
+            raise TypeError(f"unsupported operand type(s) for *: {type(self)} and {type(other)}")
+        elif (other <= 0):
+            raise ValueError(f"Non-positive multiplication factor {other} for {type(self)}")
+        else:
+            len_original = len(self)
+            offset = len(self)
+            for _ in range(other-1):
+                for i in range(len_original):
+                    self.add_module(str(i + offset), self._modules[str(i)])
+                    offset += 1
```
Comment (albanD) [inline torch/nn/modules/container.py:175]: Shouldn't this be in the outter loop and be `offset += len_original` ?

File: torch/nn/modules/container.py (hunk 3 above)
Comment (anshradh) [reply torch/nn/modules/container.py]: Yup, my bad!

File: torch/nn/modules/container.py (hunk 4 above)
Comment (anshradh) [reply torch/nn/modules/container.py]: Yes, it should!

**Review by albanD — APPROVED** [review]
Ok, the changes look good to me, let's make sure CI is green and then we can merge

**Comment by albanD** [comment]
Could you fix the lint please?

**Comment by albanD** [comment]
Ho lint is still not happy.
If you want to run it locally, you can do this from the root of the repo:
```
pip install lintrunner
lintrunner init
lintrunner -m master
```

**Comment by anshradh** [comment]
> Ho lint is still not happy. If you want to run it locally, you can do this from the root of the repo:
> 
> ```
> pip install lintrunner
> lintrunner init
> lintrunner -m master
> ```

Yeah I ran lintrunner -f and it said my local changes were fine, but looks like it wasn't able to catch everything. I'm still seeing a lint error in a file I haven't changed 'torch/utils/data/dataloader.py', not sure if that'll block merging or not.

**Pull request closed without merging**